odoo/addons/base/models/ir_rule.py

```diff
@@ -68,6 +68,18 @@ def _check_model_name(self):
         if any(rule.model_id.model == self._name for rule in self):
             raise ValidationError(_('Rules can not be applied on the Record Rules model.'))
 
+    @api.constrains('domain_force', 'model_id')
+    def _check_domain(self):
+        eval_context = self._eval_context()
+        for rule in self:
+            if rule.domain_force and rule.model_id:
+                try:
+                    # dummy search to make sure the domain is valid
+                    domain = safe_eval(rule.domain_force, eval_context)
+                    self.env[rule.model_id.model]._search(domain, limit=1)
```

Comment:
You should use sudo(), ir.rule are always check in sudo. (It may create issues with field with groups).